Mathlib/Analysis/InnerProductSpace/CanonicalTensor.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+
+import Mathlib.Analysis.InnerProductSpace.PiL2
+
+/-!
+# Canonical tensors in real inner product spaces
+
+Given an `InnerProductSpace ℝ E`, this file defines two canonical tensors.
+
+* `InnerProductSpace.canonicalContravariantTensor E : E ⊗[ℝ] E →ₗ[ℝ] ℝ`. This is
+  the element corresponding to the inner product.
+
+* If `E` is finite-dimensional, then `E ⊗[ℝ] E` is canonically isomorphic to its
+  dual. Accordingly, there exists an element
+  `InnerProductSpace.canonicalCovariantTensor E : E ⊗[ℝ] E` that corresponds to
+  `InnerProductSpace.canonicalContravariantTensor E` under this identification.
+
+The theorem `InnerProductSpace.canonicalCovariantTensorRepresentation` shows
+that `InnerProductSpace.canonicalCovariantTensor E` can be computed from any
+orthonormal basis `v` as `∑ i, (v i) ⊗ₜ[ℝ] (v i)`.
+-/
+
+open InnerProductSpace TensorProduct
+
+variable (E : Type*) [NormedAddCommGroup E] [InnerProductSpace ℝ E]
+
+/-- The canonical contravariant tensor corresponding to the inner product -/
+noncomputable def InnerProductSpace.canonicalContravariantTensor :
+    E ⊗[ℝ] E →ₗ[ℝ] ℝ := lift bilinFormOfRealInner
+
+/-- The canonical covariant tensor corresponding to `InnerProductSpace.canonicalContravariantTensor`
+under the identification of `E` with its dual -/
+noncomputable def InnerProductSpace.canonicalCovariantTensor [FiniteDimensional ℝ E] :
+    E ⊗[ℝ] E := ∑ i, ((stdOrthonormalBasis ℝ E) i) ⊗ₜ[ℝ] ((stdOrthonormalBasis ℝ E) i)
+
+/-- Representation of the canonical covariant tensor in terms of an orthonormal basis. -/
+theorem InnerProductSpace.canonicalCovariantTensorRepresentation [FiniteDimensional ℝ E]
```

Comment:
```suggestion
theorem InnerProductSpace.canonicalCovariantTensor_eq_sum [FiniteDimensional ℝ E]
```
If you change this, don't forget to change it also in the file-level docstring.

Reply:
Done. Also changed the docstring.